File: lib-google-proximity/src/main/java/com/ndipatri/iot/googleproximity/container/ObjectGraph.java
package com.ndipatri.iot.googleproximity.container;

import android.content.Context;

import com.ndipatri.iot.googleproximity.GoogleProximity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {GPModule.class})
public interface ObjectGraph {

    void inject(GoogleProximity thingy);

    final class Initializer {
        public static ObjectGraph init(final Context context, final boolean trustAllConnections) {
            return DaggerObjectGraph.builder().gPModule(new GPModule(context, trustAllConnections)).build();
        }
    }
}

File: lib-google-proximity/src/main/java/com/ndipatri/iot/googleproximity/container/ObjectGraph.kt
package com.ndipatri.iot.googleproximity.container

import android.content.Context

import com.ndipatri.iot.googleproximity.GoogleProximity

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = [GPModule::class])
interface ObjectGraph {

    fun inject(thingy: GoogleProximity)

    object Initializer {
        fun init(context: Context, trustAllConnections: Boolean): ObjectGraph {
            return DaggerObjectGraph.builder().gPModule(GPModule(context, trustAllConnections)).build()
        }
    }
}
